src/lualib-lua50/tsconfig.json

```diff
@@ -0,0 +1,21 @@
+{
+    "compilerOptions": {
+        "outDir": "../../dist/lualib-lua50",
+        "target": "esnext",
+        "lib": ["esnext"],
+        // Counterintuitively, we do not build with the 5.0 types as they are incompatible with
+        // branches written for the universal target. For now, the 5.4 types work for both targets.
+        "types": ["lua-types/5.4"],
+        "skipLibCheck": true,
+
+        "noUnusedLocals": true,
+        "noUnusedParameters": true
+    },
+    "tstl": {
+        "luaLibImport": "none",
+        "noHeader": true,
+        "luaTarget": "5.0",
+        "luaPlugins": [{ "name": "../../dist/lualib-build/plugin.js" }]
+    },
+    "include": ["../lualib", "../../language-extensions/index.d.ts"]
```

Comment:
Could you just add this file as tsconfig.lua50.json to `src/lualib` instead? I don't like this new directory

Reply:
Good idea. Implemented with https://github.com/TypeScriptToLua/TypeScriptToLua/pull/1263/commits/ced3532e7e16e9f13ce62e5f0a1b73960d1be308.